r5dev/sdklauncher/basepanel.cpp

```diff
@@ -550,30 +783,173 @@ void CUIBaseSurface::ParseMaps()
 {
 	std::regex rgArchiveRegex{ R"([^_]*_(.*)(.bsp.pak000_dir).*)" };
 	std::smatch smRegexMatches;
-	for (const auto& dEntry : fs::directory_iterator("vpk"))
-	{
-		std::string svFileName = dEntry.path().string();
-		std::regex_search(svFileName, smRegexMatches, rgArchiveRegex);
-
-		if (smRegexMatches.size() > 0)
+	if (fs::exists("vpk")) {
+		for (const auto& dEntry : fs::directory_iterator("vpk"))
 		{
-			if (strcmp(smRegexMatches[1].str().c_str(), "frontend") == 0)
-			{
-				continue;
-			}
-			else if (strcmp(smRegexMatches[1].str().c_str(), "mp_common") == 0)
+			std::string svFileName = dEntry.path().string();
+			std::regex_search(svFileName, smRegexMatches, rgArchiveRegex);
+
+			if (smRegexMatches.size() > 0)
 			{
-				if (!this->m_MapCombo->Items.Contains("mp_lobby"))
```

Comment:
The 'this->m_MapCombo->Items.Contains' has been removed here, but this has to be added back as the user will otherwise end up with duplicate map names if they have server and client VPK's installed.